src/app/zap-templates/templates/app/access.zapt

```diff
@@ -0,0 +1,420 @@
+{{> header}}
+
+// Prevent multiple inclusion
+#pragma once
+
+// Prevent changing generated format
+// clang-format off
+
+#define GENERATED_ACCESS_PRIVILEGE__VIEW (0)
+#define GENERATED_ACCESS_PRIVILEGE__OPERATE (1)
+#define GENERATED_ACCESS_PRIVILEGE__MANAGE (2)
+#define GENERATED_ACCESS_PRIVILEGE__ADMINISTER (3)
+
+////////////////////////////////////////////////////////////////////////////////
+
+{{#chip_server_clusters}}
+    {{#first}}
+// Parallel array data (*cluster*, attribute, privilege) for read attribute
+#define GENERATED_ACCESS_READ_ATTRIBUTE__CLUSTER { \
+    {{/first}}
+    {{#zcl_attributes}}
+        {{#access entity="attribute"}}
+            {{#if hasOperation}}
+                {{#if (isStrEqual operation "read")}}
+                    {{#if hasRole}}
+                        {{#if (isStrEqual role "view")}}
```

Comment:
Is there a reason `#access` does not just set operation and role to "" or something when not present so you wouldn't need to nest the checks so deeply?

Reply:
No it gets null not empty string, if we want this, I can ticket and triage it for after feature complete.

Reply:
Yep, followup please!

Reply:
Issue #16539